cpu/sensors.py

```diff
@@ -0,0 +1,116 @@
+        s_mg = SoftwareManager()
+        d_distro = distro.detect()
+        if d_distro.name == "Ubuntu":
+            if not s_mg.check_installed("lm-sensors") and not s_mg.install(
+                    "lm-sensors"):
+                self.error('Need sensors to run the test')
+        else:
+            if not s_mg.check_installed("lm_sensors") and not s_mg.install(
+                    "lm_sensors"):
+                self.error('Need sensors to run the test')
+        if d_distro.arch in ["ppc64", "ppc64le"]:
+            kernel_ver = platform.uname()[2]
+            l_config = "CONFIG_SENSORS_IBMPOWERNV"
+            config_op = process.system_output(
+                'cat /boot/config-' + kernel_ver +
+                '| grep -i --color=never ' + l_config, shell=True)
+            if "=" not in config_op:
+                self.error('Config is not set')
+            c_val = (config_op.split("=")[1]).replace('\n', '')
+            if "powerkvm" in d_distro.name:
+                if not c_val == "y":
+                    self.error('Config is not set properly')
+                else:
+                    self.log.info("Driver will be part of distro")
+            else:
+                if not c_val == "m":
+                    self.error('Config is not set correctly')
+                else:
+                    self.log.info("Driver will be built as module")
+                    mod_op = process.run(
+                        'modprobe ibmpowernv')
+                    if mod_op.exit_status == 0:
+                        lsmod_op = process.system_output(
+                            "lsmod | grep -i ibmpowernv", shell=True)
+                        if "ibmpowernv" not in lsmod_op:
+                            self.error('Module Loading Failed')
+                        else:
+                            self.log.info('Module Loaded Successfully')
+        if not d_distro.name == "Ubuntu":
+            try:
+                process.run('service lm_sensors stop', sudo=True)
+                process.run('service lm_sensors start', sudo=True)
+                process.run('service lm_sensors status', sudo=True)
+            except process.CmdError:
+                self.error(
+                    'Starting Service Failed. Make sure module is loaded')
+        process.run('sensors-detect --auto', sudo=True)
+        error_list = self.check_errors('sensors')
+        if len(error_list) > 0:
+            self.error('sensors command failed with %s' % error_list)
+        error_list = self.check_errors('sensors -f')
+        if len(error_list) > 0:
+            self.error('sensors -f command failed with %s' % error_list)
+        error_list = self.check_errors('sensors -A')
+        if len(error_list) > 0:
+            self.error('sensors -A command failed with %s' % error_list)
+        error_list = self.check_errors('sensors -u')
+        if len(error_list) > 0:
+            self.error('sensors -u command failed with %s' % error_list)
+
```

Comment:
can this self.error replaced to self.fail() as self.fail  will prompt as 'TestFail:'